### DISCUSSION TITLE
Error on pipeline

### DISCUSSION THREAD

**Discussion by valesceva** [mia-platform/community#88] [Q&A] [answered]
Good morning, developers of a client when trying to start a pipeline on gitlab project get this error: 

> Project `platform/pipelines-templates` not found or access denied! Make sure any includes in the pipeline configuration are correctly defined.

Do you know how we can solve it?

**Comment by frengooo** [comment, marked as answer]
Hi @valesceva, you probably just need to add the developers of your client as "Reporter" to the `platform/pipelines-templates` repository.
They need to access that repository to execute our pipelines.

**Reply by theliuk** [reply]
What if the those developers are outsiders? They are not Mia's developers.

**Reply by frengooo** [reply]
If they need to run CI pipelines, or to deploy using `mlp`, they must have permissions on that repo.
We already have a lot of outsiders, but with a low role (usually "Reporter").

**Reply by valesceva** [reply]
We don't have permission to add users to repo. Someone could give to me or @theliuk a role that have such permissions?

**Reply by theliuk** [reply]
@EdoCrippaOfficial proposed to fork the repo inside our customer group.

**Reply by frengooo** [reply]
I don't know if this can really be a solution. If we fork the repo inside all customers groups we need to keep them updated and still we need to add the users to be able to use pipelines.
Moreover, we need to edit all the references on the `.gitlab-ci` files, to let them use the new pipelines repository.'

I think that all TLs should have permissions to add members as `Reporter` to that repository.

**Reply by edocrippaofficial** [reply]
Yes, I see a scalability problem if we start to add all customers developers as guest to our platform project.
Instead, we could fork the platform project `pipelines-templates` in the client repo, and assign the developers to that forked repo.
The only thing is that it has to be manually re-aligned to the original repo once in a while... But this is a minor downside overall
The reference edit on the `gitlab-ci` has to be done only once
Let me know what you think

**Reply by theliuk** [reply]
I agree with both of you, but more with @EdoCrippaOfficial :) . 
`platform/pipeline-templates` should be a template project, which can be used directly by Mia core services.

Forking is a good idea when dealing with a client, because you want to have total control. 
By doing so, you can also migrate the whole client group to another gitlab and everything would work fine.

**Reply by frengooo** [reply]
Ok, so we should do it for all clients, not only for one, but we need to define a flow to keep the clients repositories updated.
@davidebianchi @fredmaggiowski What do you think about this proposal?

**Reply by fredmaggiowski** [reply]
Gitlab forks should have features to help in being aligned with the original repo so this could be feasible with the tools managed by us

**Reply by edocrippaofficial** [reply]
https://about.gitlab.com/blog/2016/12/01/how-to-keep-your-fork-up-to-date-with-its-origin/

**Reply by theliuk** [reply]
<img width="1012" alt="Schermata 2022-11-11 alle 12 58 16" src="https://user-images.githubusercontent.com/809858/201335970-ab247587-632b-4bd0-abbb-d5f2e07caca4.png">

**Reply by davidebianchi** [reply]
Permissions should be given at user registration. If this is not the case, it's an issue when the new user is added.

If you make a mirror, you should not modify the repo so the fork is not very effective. Furthermore, from the marketplace will be set the common pipelines, and will not use the group template so at the moment is not possible to do it (but yes, it can be an interesting improvement to add)

**Reply by theliuk** [reply]
So, I forked the repo and I'm not going to modify it.

**Reply by frengooo** [reply]
Is this discussion solved? @theliuk @valesceva

**Reply by theliuk** [reply]
Yes

**Comment by calairoh** [comment]
Hi, I'll go back to the previous thread because we had the same problem. I already read the previous discussion and I understand the two ways. However, I have a question: could be possible to let "Pipeline Template" public available? If not, why? Is there any critical information in that repo?

**Reply by fredmaggiowski** [reply]
As of now it is not possible to make pipelines templates public; we are actually working on a whole new set of pipelines that will be public on GitHub so it's a matter of time (it may still take some time since there is **a lot** of work to do). But we are getting this covered 💪

**Reply by calairoh** [reply]
Great to hear it! Thanks!
